typescript/src/lib/utils/ledger.ts

```diff
@@ -0,0 +1,177 @@
+import { ethers, Signer } from "ethers"
+import {
+  Account,
+  WalletAPIClient,
+  WindowMessageTransport,
+} from "@ledgerhq/wallet-api-client"
+import BigNumber from "bignumber.js"
+import { AddressZero } from "@ethersproject/constants"
+import { Deferrable } from "@ethersproject/properties"
+import { Hex } from "./hex"
+class AccountNotFoundError extends Error {
+  constructor() {
+    super(
+      "Account not found. Please use `requestAccount` method or set the signer account with `setAccount` method."
+    )
+  }
+}
+
+export class LedgerLiveAppEthereumSigner extends Signer {
+  private _walletApiClient: WalletAPIClient
+  private _windowMessageTransport: WindowMessageTransport
+  private _account: Account | undefined
+
+  constructor(
+    provider: ethers.providers.Provider,
+    windowMessageTransport?: WindowMessageTransport,
+    walletApiClient?: WalletAPIClient
+  ) {
+    super()
+    ethers.utils.defineReadOnly(this, "provider", provider || null)
+    this._windowMessageTransport =
+      windowMessageTransport || getWindowMessageTransport()
+    this._walletApiClient =
+      walletApiClient || getWalletAPIClient(this._windowMessageTransport)
+  }
+
+  get account() {
+    return this._account
+  }
+
+  setAccount(account: Account | undefined): void {
+    this._account = account
+  }
+
+  async requestAccount(
+    params: { currencyIds?: string[] | undefined } | undefined
+  ): Promise<Account> {
+    this._windowMessageTransport.connect()
+    const account = await this._walletApiClient.account.request(params)
+    this._windowMessageTransport.disconnect()
+    this._account = account
+    return this._account
+  }
+
+  getAccountId(): string {
+    if (!this._account || !this._account.id) {
+      throw new AccountNotFoundError()
+    }
+    return this._account.id
+  }
+
+  async getAddress(): Promise<string> {
+    if (!this._account || !this._account.address) {
+      throw new AccountNotFoundError()
+    }
+    return this._account.address
```

Comment:
Are we sure it's a checksum address? Ethers expect that.

Reply:
I've checked it and I confirm that this returns checksum address.